### PR TITLE
Added getLedger, getBalances and getTx API calls

### PR DESCRIPTION
**Pull request by vahaknp**

### CODE REVIEW

File: api/routes/accountBalances.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  /**
+  * getBalances
+  * use ledger_index from getLedger api call
+  * to get balances using ripple REST 
+  */
+ function getBalances(ledger_index, account) {
+    if (!account) errorResponse({error: 'Must provide account.', code:400});
+    var url = 'https://api.ripple.com/v1/accounts/'+account+'/balances';
```
Comment (sublimator) [inline api/routes/accountBalances.js]: Is this a stopgap?  client -> history -> rest -> rippled ?

File: api/lib/db.js [hunk 2]
```diff
@@ -32,6 +32,173 @@ var DB = function(config) {
...
+        for (var i=0; i<transactions.length; i++){
+          transaction_list.push(transactions[i].tx_hash);
+        }
+        ledger.transactions = transaction_list;
+      }
+      else if (options.tx_return === "binary") ledger.transactions = transactions;
+      else if (options.tx_return === "json") {
+        for (var i=0; i<transactions.length; i++){
+          var row = transactions[i];
+          row.tx = new SerializedObject(row.tx_raw).to_json();
```
Comment (sublimator) [inline api/lib/db.js]: No `hash` or `ledger_index` etc? Could probably extract a row -> json helper from some of the code in the account_tx handler and use it for the ledger transactions and transaction by hash ?

Comment (sublimator) [inline api/lib/db.js]: Or am I reading this wrong? You're sending the row directly back, which already has some fields?

Comment (sublimator) [inline api/lib/db.js]: No, seems to be missing some fields

File: api/routes/accountBalances.js (hunk 1 above)
Comment (shekenahglory) [inline api/routes/accountBalances.js]: We wanted the ability to get an account balance by date - the history API allows us to look up a ledger_index by date, which we can send via ripple rest to get the account balances at that ledger formatted for accounting purposes.

File: api/lib/db.js (hunk 2 above)
Comment (shekenahglory) [inline api/lib/db.js]: I think we want to standardize all the transaction responses - there are slight differences for each from rippled's account_tx, ledger, and tx.  This is what I propose:

Transaction
	•	tx
	⁃	raw tx fields
	•	meta
	⁃	metadata fields
	•	hash
	•	date (unix)
	•	ledger_index

Transaction (binary)
	•	tx (binary)
	•	meta (binary)
	•	hash
	•	date (unix)
	•	ledger_index

**Comment by shekenahglory** [comment]
I'm going to merge this as is, but we do need to look at the format and fields in the responses

**Pull request merged**